### PR TITLE
CC-489 - Infra updates for the Node 18 upgrade

### PR DESCRIPTION
**Pull request by Zain-Abbas1**
Jira (has relevant sub-tasks: CC-692, CC-693, CC-694 & CC-696) - https://companieshouse.atlassian.net/browse/CC-489

- Add ecs-image-build folder
- Add terraform/groups/ecs-service folders
- Add vault-providers folder
- Change Makefile test target

### CODE REVIEW

File: terraform/groups/ecs-service/main.tf [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+provider "aws" {
+  region  = var.aws_region
+}
+
+terraform {
+  backend "s3" {}
+}
+
+module "secrets" {
+  source = "git@github.com:companieshouse/terraform-modules//aws/ecs/secrets?ref=1.0.191"
```
Comment (markpit) [inline terraform/groups/ecs-service/main.tf:24]: for extensions-web , we were asked to change the "secrets" source line to a newer version
  source = "git@github.com:companieshouse/terraform-modules//aws/ecs/secrets?ref=1.0.192"

and for the "ecs-service" source - 
  source = "git@github.com:companieshouse/terraform-modules//aws/ecs/ecs-service?ref=1.0.192"

**Comment by mwestacott** [comment]
@Zain-Abbas1 please could you add JIRA link to the main commit message - several probably, in this case, but normally we'd have one PR per sub-task (makes reviews simpler).

Also, you probably need to mark those other sub-tasks in JIRA as in-progress.

File: Makefile [hunk 2]
```diff
@@ -15,8 +15,11 @@ test:
 	npm run test:coverage
 
 .PHONY: test-unit
-test-unit:
-	npm run test
+test-unit: clean
+	# 'ECS pipeline calls test-unit so need to run sonar-test here'
+	npm run sonar-test
+#test-unit:
```
Comment (mwestacott) [inline Makefile:21]: Commented out lines here can be deleted.

File: src/config.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ const config: ApplicationConfiguration = {
         cacheServer: process.env.CACHE_SERVER as string
     },
     applicationNamespace: "restricted-word-web",
-    baseUrl: process.env.CHS_URL ?? "https://chs.local"
+    baseUrl: process.env.CHS_URL ?? "http://chs.local"
```
Comment (mwestacott) [inline src/config.ts:16]: Wondering about this change. Why was it done?

Possibly this is PA's change so would be good to get that merge of 'master' into node 18 branch to make reviewing this easier.

Comment (Zain-Abbas1) [reply src/config.ts]: I've updated the base branch with master and this has now disappeared.

File: terraform/groups/ecs-service/locals.tf [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Define all hardcoded local variable and local variables looked up from data resources
+locals {
+  stack_name                = "search" # this must match the stack name the service deploys into
```
Comment (dlloyd1-ch) [inline terraform/groups/ecs-service/locals.tf:3]: Is this correct? The other service already live in this cluster ('search-web') has a 'stack_name' of 'search-service' - https://github.com/companieshouse/search.web.ch.gov.uk/blob/a4f855222f4ec0b11ff2a7cecb18b55aa251d788/terraform/groups/ecs-service/locals.tf#L3

File: Makefile (hunk 2 above)
Comment (Zain-Abbas1) [reply Makefile]: Done

File: terraform/groups/ecs-service/locals.tf (hunk 4 above)
Comment (Zain-Abbas1) [reply terraform/groups/ecs-service/locals.tf:3]: I had it as `search-service` before but was told to remove service as something similar was done for extensions. If `search-web` has already gone live with it perhaps that is the correct way of doing it then, can we confirm with someone?

Comment (tobafolaranmi-ch) [reply terraform/groups/ecs-service/locals.tf:3]: @dlloyd1-ch is correct. It is `search-service` in this case

Comment (Zain-Abbas1) [reply terraform/groups/ecs-service/locals.tf]: Done

**Review by mwestacott — APPROVED** [review]
Looks fine to me. Would check with @dlloyd1-ch before merging though as he has an unresolved comment.

File: terraform/groups/ecs-service/locals.tf [hunk 5]
```diff
@@ -0,0 +1,55 @@
+# Define all hardcoded local variable and local variables looked up from data resources
+locals {
+  stack_name                = "search-service" # this must match the stack name the service deploys into
+  name_prefix               = "${local.stack_name}-${var.environment}"
+  service_name              = "restricted-word-web"
+  container_port            = "3000" # default node port required here until prod docker container is built allowing port change via env var
+  docker_repo               = "restricted-word-web"
+  lb_listener_rule_priority = 96
+  lb_listener_paths         = ["/admin/restricted-word","/admin/restricted-word/.*%"]
```
Comment (tobafolaranmi-ch) [inline terraform/groups/ecs-service/locals.tf:9]: This path won't work `"/admin/restricted-word/.*%"`, regex syntax is different on aws use `"/admin/restricted-word/*"`

Comment (Zain-Abbas1) [reply terraform/groups/ecs-service/locals.tf]: Done

**Review by tobafolaranmi-ch — APPROVED** [review]
LGTM

**Review by dlloyd1-ch — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**